src/sqlfluff/rules/L052.py

```diff
@@ -193,21 +193,38 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
                     anchor_segment = self._handle_trailing_inline_comments(
                         context, anchor_segment
                     )
-
-                    fixes = [
-                        LintFix.replace(
-                            anchor_segment,
-                            [
+                    fixes = []
+                    if anchor_segment == context.segment:
+                        fixes.append(
+                            LintFix.replace(
                                 anchor_segment,
-                                NewlineSegment(),
-                                SymbolSegment(raw=";", type="symbol", name="semicolon"),
-                            ],
-                        ),
-                        LintFix.delete(
-                            context.segment,
-                        ),
-                    ]
-                    fixes.extend(LintFix.delete(d) for d in whitespace_deletions)
+                                [
+                                    NewlineSegment(),
+                                    SymbolSegment(
+                                        raw=";", type="symbol", name="semicolon"
+                                    ),
+                                ],
+                            )
+                        )
+                    else:
+                        fixes.extend(
+                            [
+                                LintFix.replace(
```

Comment:
It looks like the two occurrences of `LintFix.replace()` are identical. Could we pull that part of the fix out of the `if` to avoid duplicating that code?

Reply:
They're not the same:
first one:
```python
LintFix.replace(
                                anchor_segment,
                                [
                                    NewlineSegment(),
                                    SymbolSegment(
                                        raw=";", type="symbol", name="semicolon"
                                    ),
                                ],
                            )
```

second one:
```python
LintFix.replace(
                                anchor_segment,
                                [
                                    anchor_segment,
                                    NewlineSegment(),
                                    SymbolSegment(
                                        raw=";", type="symbol", name="semicolon"
                                    ),
                                ],
                            )
```

Second one keeps the anchor segment since it's not the semicolon and shouldn't be overwritten